src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=ungrouped-imports
+# Standard
+from datetime import datetime
+from typing import List
+import enum
+import logging
+import os
+import pathlib
+
+# Third Party
+from instructlab.eval.mt_bench_common import (
+    get_openai_client as get_local_openai_client,
+)
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator
+from openai import OpenAI, OpenAIError
+from ragas.evaluation import EvaluationResult  # type: ignore
+import pandas as pd
+
+# First Party
+from instructlab import client_utils
+
+# Local
+from ..configuration import DEFAULTS, _serve
+from .evaluate import get_model_name as get_local_model_name
+from .evaluate import launch_server, validate_model
+
+logger = logging.getLogger(__name__)
+
+
+class IOFileType(enum.Enum):
+    CSV = "csv"
+    JSONL = "jsonl"
+    XLSX = "xlsx"
+
+
+def validate_input_questions(input_questions: pathlib.Path):
+    """
+    Ensure input questions file exists, is not a directory
+    and is a '.jsonl' file.
+    Args:
+        input_questions (Path): The path to the input questions .jsonl file
+    Returns:
+        None
+    """
+    if not os.path.exists(input_questions):
+        raise ValueError(
+            f"Input questions file {input_questions} does not exist",
+        )
+
+    if os.path.isdir(input_questions):
+        raise ValueError(
+            f"Input questions file {input_questions} is a directory",
+        )
+
+    # need to make sure input questions is a jsonl file
+    if input_questions.suffix.lstrip(".") != IOFileType.JSONL.value:
+        raise ValueError(
+            f"Invalid file type: {input_questions}. Expected a '.jsonl' file."
+        )
+
+
+def validate_output_file_formats(file_formats: List[str]):
+    """
+    Validates that file formats passed in for the file containing scores and
+    responses is a valid output format. Valid file formats are one of:
+    jsonl, xlsx, or csv.
+    Args:
+        file_formats (List[str]): A list of all of the file format strings
+                                  passed in by the user.
+    Returns:
+        None
+    """
+    for file_format in file_formats:
+        valid_formats = {item.value for item in IOFileType}
+        if file_format not in valid_formats:
```

Comment:
FYI you can reduce this logic like this:

```suggestion
        if not any (file_format == item.value for item in IOFileType):
```